### PR TITLE
[WIP] [air] Allow users to use instances of `ScalingConfig`

### PR DESCRIPTION
**Pull request by sumanthratna**
CC @amogkam @xwjiang2010 @richardliaw 

TODO
- [ ] update ipynb examples
- [ ] Tuner needs to support this PR (i.e., test_tuner should currently fail)
- [ ] update tune/train docs 

## Why are these changes needed?

Makes `ScalingConfig` the single way to configure scaling configurations for Tune

## Related issue number

n/a

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by ericl — CHANGES_REQUESTED** [review]
Can you ensure we have a test with a tune.grid_search on num_workers and that it is working the same as the {"num_workers": tune.grid_search()}?

File: python/ray/tune/tests/test_tuner.py [hunk 1]
```diff
@@ -91,9 +91,7 @@ def test_tuner_with_xgboost_trainer(self):
         # prep_v1 = StandardScaler(["worst radius", "worst area"])
         # prep_v2 = StandardScaler(["worst concavity", "worst smoothness"])
         param_space = {
-            "scaling_config": {
-                "num_workers": tune.grid_search([1, 2]),
-            },
+            "scaling_config": ScalingConfig(num_workers=tune.grid_search([1, 2])),
```
Comment (ericl) [inline python/ray/tune/tests/test_tuner.py:94]: Can we document this capability in Tune and in Train docs?

**Comment by sumanthratna** [comment]
> Can you ensure we have a test with a tune.grid_search on num_workers and that it is working the same as the {"num_workers": tune.grid_search()}?

@ericl Not sure I understand -- don't we already have a unit test to verify that this works? (the same test that you commented on in your PR review)
```python
Tuner(
    param_space={
        "scaling_config": ScalingConfig(num_workers=tune.grid_search())
    }
)
```

**Comment by sumanthratna** [comment]
superseded by https://github.com/ray-project/ray/pull/25712

**Pull request closed without merging**